In [1]:
import matplotlib.pyplot as plt
import numpy as np

import pandas as pd
import scipy.stats as stats

### Задача 1
Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):

zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],

ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].

Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.

Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [2]:
zp = np.array ([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])

ks = np.array ([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

In [4]:
cov = np.mean(zp*ks) - np.mean(zp) * np.mean(ks) 
cov

9157.839999999997

In [7]:
np.cov(zp, ks, ddof = 0)

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

Если не прописать аргумент ddof = 0 (смещенная ковариация) в np.cov(zp, ks), то возьмется по умолчанию ddof = 1 и будет несмещенная ковариация со значением = 10175.37777778

Найдем коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков:

In [14]:
r = cov/(np.std(zp)*np.std(ks))
r

0.8874900920739158

Найдем коэффициент корреляции Пирсона с помощью функции

In [15]:
np.corrcoef(zp, ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

Коэффициент корреляции Пирсона равен 0,88, что свидетельствует о сильной прямой линейной зависимости.

### Задача 2
Измерены значения IQ выборки студентов, обучающихся в местных технических вузах:

131, 125, 115, 122, 131, 115, 107, 99, 125, 111.

Известно, что в генеральной совокупности IQ распределен нормально.

Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [18]:
X=np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])

In [19]:
n=len(X)
n

10

In [20]:
X_mean=np.mean(X)
X_mean

118.1

In [21]:
X_var = np.var(X, ddof=1)
X_var

111.21111111111111

In [25]:
t1=stats.t.ppf(0.975, n-1)
t1

2.2621571627409915

In [26]:
int_ver = X_mean + t1*np.sqrt(X_var/n)
int_ver

125.64391634841274

In [27]:
int_niz = X_mean - t1*np.sqrt(X_var/n)
int_niz

110.55608365158724

Доверительный интервал для математического ожидания [110.55; 125.64]

### Задача 4 
Известно, что рост футболистов в сборной распределен нормально с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27, среднее выборочное составляет 174.2. 
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [28]:
dispersia = 25
n = 27
X_mean = 174.2

In [31]:
alfa_popolam_5 = stats.norm.ppf(1-0.05/2)
alfa_popolam_5

1.959963984540054

In [32]:
int_ver = X_mean + alfa_popolam_5*np.sqrt(dispersia/n)
int_ver

176.08597622346028

In [33]:
int_niz = X_mean - alfa_popolam_5*np.sqrt(dispersia/n)
int_niz

172.3140237765397

Доверительный интервал для математического ожидания [172.31; 176.08]